src/librustc_mir/transform/mod.rs

```diff
@@ -98,14 +104,29 @@ fn mir_keys(tcx: TyCtxt<'_>, krate: CrateNum) -> &DefIdSet {
 }
 
 /// Where a specific `mir::Body` comes from.
-#[derive(Debug, Copy, Clone)]
+#[derive(Copy, Clone)]
 pub struct MirSource<'tcx> {
     pub instance: InstanceDef<'tcx>,
 
     /// If `Some`, this is a promoted rvalue within the parent function.
     pub promoted: Option<Promoted>,
 }
 
+impl std::fmt::Debug for MirSource<'tcx> {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        ty::tls::with(|tcx| {
+            let substs = InternalSubsts::identity_for_item(tcx, self.def_id());
+            FmtPrinter::new(tcx, &mut *f, Namespace::ValueNS)
+                .print_def_path(self.def_id(), substs)?;
```

Comment:
Wouldn't it be better to make a fake `Instance` and print that?
IIRC, `Instance` prints shims specially, or at least I think it does?

Reply:
it does, i'll try that

Reply:
see below, I think we should revert all that and do it separately in another PR since it has regression implications that I don't want to mix with this PR